### PR TITLE
Dev/prechecks

### PR DESCRIPTION
**Pull request by micahwoodard**
Add all check function scattered through out spim repos and added a check read/write function as well that uses fios

### CODE REVIEW

**Comment by Poofjunior** [comment]
@micahwoodard it looks like some of the "local=external directory" code logic made it into this branch. Let's first merge #20 then merge #18. After that, let's update this branch against main, and those extra changes should disappear. If that works, we should be able to merge this too afterwards.

File: spim_core/spim_base.py [hunk 1]
```diff
@@ -112,6 +114,95 @@ def log_to_file(self, filepath: Path, logger: Logger = None,
...
             logger.removeHandler(log_handler)
 
+    def check_ext_disk_space(self, xtiles, ytiles, ztiles):
+        """Checks ext disk space before scan to see if disk has enough space scan
+
+        :param xtiles: number of x tiles
+        :param ytiles: number of y tiles
+        :param ztiles: number of z tiles
+        """
+        # One tile (tiff) is ~10368 kb
```
Comment (adamkglaser) [inline spim_core/spim_base.py:124]: Can get rid of this comment

File: spim_core/spim_base.py [hunk 2]
```diff
@@ -112,6 +114,95 @@ def log_to_file(self, filepath: Path, logger: Logger = None,
...
+            if speed_MB_s['write'] <= acq_speed_MB_s:
+                write_too_slow = True
+                self.log.warning(f'{drive} write speed too slow')
+
+            if read_too_slow or write_too_slow:
+                raise
+        except subprocess.CalledProcessError:
+            self.log.warning('fios not installed on computer. Cannot verify read/write speed')
+
+    def _check_system_memory_resources(self, channel_count: int,
```
Comment (adamkglaser) [inline spim_core/spim_base.py:184]: Is there a reason this function and not the others are _function()?

File: spim_core/spim_base.py [hunk 3]
```diff
@@ -112,6 +114,95 @@ def log_to_file(self, filepath: Path, logger: Logger = None,
...
+
+           :param z_tiles: number of z tiles
+        """
+
+        est_filesize = self.cfg.bytes_per_image * z_tiles
+        if est_filesize * 2 >= shutil.disk_usage(self.cfg.local_storage_dir).free:
+            self.log.error("Not enough space on disk. Is the recycle bin empty?")
+            raise
+
+    def check_read_write_speeds(self, drive: Path):
```
Comment (adamkglaser) [inline spim_core/spim_base.py:142]: Just so we remember what all of the fio inputs are. I think we should pull --size, --bs, --direct, --numjobs, etc. out into input variables, and document them in this function. We can make the defaults if not specified what the current values are. We should also add a flag for linux, which would change the ioengine to the default linux one which is 'libaio'.

**Review by adamkglaser — CHANGES_REQUESTED** [review]
Overall looks good to me, just a few comments.

File: spim_core/spim_base.py (hunk 2 above)
Comment (micahwoodard) [reply spim_core/spim_base.py:184]: Sonya wrote this one so I'm not entirely sure her reasoning on making this function private

**Pull request merged**